### PR TITLE
Optional to have a config

### PR DESCRIPTION
**Pull request by TimonPost**

### CODE REVIEW

File: src/net/socket.rs [hunk 1]
```diff
@@ -49,6 +51,13 @@ impl Socket {
         ))
     }
 
+    /// Configure the socket with the passed configuration.
+    pub fn with_config(mut self, config: Config) -> Socket {
```
Comment (jstnlef) [inline src/net/socket.rs]: I think I would prefer to have a `bind_with_config` method which is similar to the bind method today, as well as a `bind` method which calls it with `Config::default()`

File: examples/server_client.rs [hunk 2]
```diff
@@ -8,8 +8,10 @@ use std::thread;
 
 const SERVER: &str = "127.0.0.1:12351";
 
+fn test() {}
```
Comment (LucioFranco) [inline examples/server_client.rs]: whats this for?

Comment (TimonPost) [reply examples/server_client.rs]: I taught it looked pretty!, no just kidding, it was from an other PR I was planning to make. But it should be removed.

File: examples/udp.rs [hunk 3]
```diff
@@ -22,12 +22,8 @@ fn server_address() -> SocketAddr {
 
 /// This is an example of how to send data to an specific address.
 pub fn send_data() {
-    // you can change the config but if you want just go for the default.
-    let config = Config::default();
-
     // setup an udp socket and bind it to the client address.
```
Comment (fhaynes) [inline examples/udp.rs]:
```suggestion
    // Setup a udp socket and bind it to the client address.
```

File: examples/simple_udp.rs [hunk 4]
```diff
@@ -81,12 +81,8 @@ struct Server {
 impl Server {
     #[allow(unused_must_use)]
```
Comment (fhaynes) [inline examples/simple_udp.rs]: Can we remove this?

File: examples/server_client.rs [hunk 5]
```diff
@@ -9,7 +9,7 @@ use std::thread;
 const SERVER: &str = "127.0.0.1:12351";
 
 fn server() -> Result<(), ErrorKind> {
-    let (mut socket, packet_sender, event_receiver) = Socket::bind(SERVER, Config::default())?;
+    let (mut socket, packet_sender, event_receiver) = Socket::bind(SERVER)?;
     let _thread = thread::spawn(move || socket.start_polling());
 
     println!("Listening for connections to {}", SERVER);
```
Comment (fhaynes) [inline examples/server_client.rs]: We should probably use a logging facility for this, `info!()` for example

File: examples/simple_udp.rs [hunk 6]
```diff
@@ -149,12 +145,8 @@ struct Client {
 impl Client {
     #[allow(unused_must_use)]
```
Comment (fhaynes) [inline examples/simple_udp.rs]: Do we need this?

File: examples/server_client.rs (hunk 5 above)
Comment (TimonPost) [reply examples/server_client.rs]: It is not related to my PR but I can throw it in.

Comment (fhaynes) [reply examples/server_client.rs]: Apologies, didn't even think about that. It's fine with me with either way. Resolving this.

**Pull request merged**